src/shared/components/query/QueryStore.ts

```diff
@@ -1511,6 +1511,16 @@ export class QueryStore {
         return new Set(referenceGenomes);
     }
 
+    @computed get readPermissions(): Set<string> {
+        const studies = Array.from(this.treeData.map_node_meta.keys()).filter(
+            s => typeof (s as CancerStudy).readPermission !== 'undefined'
```

Comment:
```(s:CancerStudy)=>s.readPermission===undefined```
should work.  You don't need typeof anymore.  Possible that you will need to cast with as, but not sure.